ironic/drivers/modules/drac/raid.py

```diff
@@ -924,13 +918,13 @@ def delete_configuration(self, task):
         node_state = _commit_to_controllers(node, controllers)
         LOG.info("Converting all physical disks to JBOD mode")
         change_state_result = configure_physical_disk_conversion(node,
-            mode="JBOD")
+                                                                 mode="JBOD")
 
         if change_state_result:
             LOG.info("Completed converting all physical disks to jbod mode")
             return None
-        return node_state
 
+        return node_state
```

Comment:
The reason i am returning None in above line i  am explaining below:
1. Convert all disk to JBOD(non-raid) mode, after deleting raid(VD, hotspares/dedicated hotspares).
2. Now currently `delete_configuration` function returns `clean_wait` after deleting virtual disks , its resumes cleaning and assign_roles converts all disks to jbod.
-------------Here comes new implementation-------------------
3. Now modifying `delete_configuration` call by applying  `reset_configuration` , it returns the same `clean_wait` but unfortunately it keeps giving node state in `clean_wait` instead of  resuming cleaning and then `manageable`.
3. So after reset_configuration call i applied call for conversion of disks to raid 
and when i return None from `delete_configuration()`,  it successfully converts all physical disks to `NON-RAID` 
4. I am not sure whether to return `None` or `clean_wait`
NOTE:  `clean_wait` returns by `_commit_to_controllers()`
Please let me know your review.

Reply:
As I understand it, you return CLEANWAIT if an operation executes asynchronously.  Meaning that the operation continues execution after the method returns.  You only return None if the operation has completed.
I'll look into this more on Monday.

Reply:
Yes actually, reset_configuration() and disk_conversion() call cannot call asynchronously.
because reset_configuration should be completed first and after that only i can check all disks which are not in NON-RAID and can convert into NON-RAID/JBOD mode